prebackup/prebackup.go

```diff
@@ -119,64 +122,98 @@ func (p *PreBackup) generateDeployments(templates *k8upv1alpha1.PreBackupPodList
+	namespace := deployment.GetNamespace()
+	p.Log.Info("starting pre backup pod", "namespace", namespace, "name", name)
+
+	err = p.Client.Create(p.CTX, &deployment)
+	deploymentExists := errors.IsAlreadyExists(err)
+	if err != nil && !deploymentExists {
+		err := fmt.Errorf("error creating pre backup pod '%v/%v': %w", namespace, name, err)
+		p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonCreationFailed, err.Error())
+		return err
+	}
+
+	if deploymentExists {
+		err := p.Client.Get(p.CTX, types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, &deployment)
 		if err != nil {
-			err := fmt.Errorf("could not create watcher for '%v/%v': %w", namespace, name, err)
-			p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonCreationFailed, err.Error())
+			err := fmt.Errorf("error getting pre backup pod '%v/%v': %w", namespace, name, err)
+			p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonRetrievalFailed, err.Error())
 			return err
 		}
 
-		err = p.waitForReady(watcher)
+		ready, err := p.isDeploymentReady(&deployment)
 		if err != nil {
-			err := fmt.Errorf("error during deployment watch of '%v/%v': %w", namespace, name, err)
-			p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonFailed, err.Error())
+			err := fmt.Errorf("error checking the readyness of deployment '%s/%s': %s", deployment.Namespace, deployment.Name, err)
+			p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonRetrievalFailed, err.Error())
 			return err
 		}
+		if ready {
+			p.Log.V(2).Info("pre backup pod already in ready state", "deployment", fmt.Sprintf("%s/%s", deployment.Namespace, deployment.Name), "event type")
+			p.SetConditionTrue(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonSucceeded)
+			return nil
+		}
 	}
 
-	p.SetConditionTrue(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonSucceeded)
+	watcher, err := clientset.AppsV1().
+		Deployments(deployment.GetNamespace()).
+		Watch(p.CTX, metav1.SingleObject(deployment.ObjectMeta))
+	if err != nil {
+		err := fmt.Errorf("could not create watcher for '%v/%v': %w", namespace, name, err)
+		p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonCreationFailed, err.Error())
+		return err
+	}
+
+	err = p.waitForReady(watcher)
+	if err != nil {
+		err := fmt.Errorf("error during deployment watch of '%v/%v': %w", namespace, name, err)
+		p.SetConditionFalseWithMessage(ConditionPreBackupPodsReady, k8upv1alpha1.ReasonFailed, err.Error())
+		return err
+	}
 	return nil
 }
 
-func (p *PreBackup) getClientset() (*kubernetes.Clientset, error) {
+func getKubeConfig() (*rest.Config, error) {
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	restConfig, err := kubeconfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
might as well return directly, right?
```suggestion
	return kubeconfig.ClientConfig()
```
if not, `return restconfig, err` should do it